framework/helpers/BaseJson.php

```diff
@@ -151,9 +151,17 @@ protected static function processData($data, &$expressions, $expPrefix)
                 $expressions['"' . $token . '"'] = $data->expression;
 
                 return $token;
-            } elseif ($data instanceof \JsonSerializable) {
+            }
+
+            if ($data instanceof \JsonSerializable) {
                 return static::processData($data->jsonSerialize(), $expressions, $expPrefix);
-            } elseif ($data instanceof Arrayable) {
+            }
+
+            if ($data instanceof \DateTimeInterface) {
```

Comment:
DateTimeInterface appeared only in 5.5 version (https://www.php.net/manual/en/class.datetimeinterface.php)
So it breaks BC, because Yii2 supports minimal 5.4 version https://github.com/yiisoft/yii2/blob/2.0.31/composer.json#L71

Reply:
`instanceof` works fine with non-existing types - you don't need to have `DateTimeInterface`  to know that `$data` is not `DateTimeInterface`.

Reply:
Since the issue doesn't affect PHP versions lower than 7.4, non-existence of `\DateTimeInterface` is fine.